go/runtime/nodes/vresioned.go

```diff
@@ -280,15 +247,24 @@ func (nw *nodeDescriptorWatcher) worker(ch <-chan *registry.NodeEvent, sub pubsu
 	}
 }
 
-// NewNodeDescriptorWatcher creates a new node descriptor watcher.
-func NewNodeDescriptorWatcher(ctx context.Context, registry registry.Backend) (NodeDescriptorWatcher, error) {
+func (nw *versionedNodeDescriptorWatcher) Versioned() bool {
+	// This watcher supports versions.
+	return true
+}
+
+// NewBaseVersionedNodeDescriptorWatcher creates a new base versioned node descriptor watcher.
+//
+// The base watcher will only track nodes that will be explicitly marked to watch
+// via WatchNode/WatchNodeWithTags methods.
+// TODO: better name?
```

Comment:
Is there a non-base version? Maybe just remove the "Base"?

Reply:
Yeah i agree that `VersionedNodeDescriptorWatcher` is better without `Base`. I thought about including something in the name that indicates that this watcher doesn't actually watch anything, and kinda serves as a "base" for a concrete watcher (e.g. the committee watcher). 
But now i also added the docstring that this mentions that this watcher only tracks nodes explicitly marked, which i think is good enough and can drop the `Base` from the name.